### ISSUE TITLE
NeoMutt should report errors when sendmail fails

### ISSUE THREAD

**Issue by protist** [neomutt/neomutt#1309]
## What would you like to be able to do?

Recently, there were [two unrelated bugs](https://github.com/neomutt/neomutt/issues/1295) that caused piping to an external sendmail to fail. NeoMutt reported no errors, and this failure was not reported. In both cases, I only discovered this a few days later, when a contact complained that the email I had sent never arrived. By this point, I though I had sent several emails to various contacts.

I could only identify this failure because I use a [custom sendmail script](https://unix.stackexchange.com/questions/240885/can-i-send-email-in-the-background-in-mutt-while-still-being-notified-of-errors/241632#241632) that logs its activities, then passes the email on to `msmtp`. Users that directly use `msmtp` or other sendmails would have no way to knowing if their emails had been unsent.

## What needs to change in NeoMutt?

[Ideally](https://github.com/neomutt/neomutt/issues/1295#issuecomment-410434207), NeoMutt should report an error when it fails to send email.

## Have you tried any alternative methods?

No

**Comment by roccoblues** [comment]
Trying to cleanup some older issues here. Can anyone comment if this issue still persists with the latest version? Thanks!

**Comment by protist** [comment]
Yes, this issue is still present in Neomutt 20231103. Thanks for checking @roccoblues !

**Comment by roccoblues** [comment]
I tried to reproduce this but so far couldn't. Here's what I tried:

### Wrong sendmail command ###

`set sendmail = "foo"`

-> NeoMutt reports that it failed to spawn the command.

### Broken `msmtp` config ###

`host foo.my.domain` in `~/.msmtprc`

-> NeoMutt reports `Error sending message, child exited 68 (Host unknown.)`

and

```
msmtp: cannot locate host foo.my.domain: nodename nor servname provided, or not known
msmtp: could not send mail (account my-account from /Users/dennis/.msmtprc)
```

@protist can you provide more information how to reproduce this? A minimal config example would be great.

**Comment by protist** [comment]
Thanks @roccoblues . I think possibly it's related to having `set sendmail_wait=-1`?

I tested with this set, and I get the issue as described. For example, `set sendmail = "foo"` fails silently.

However, if I set `set sendmail_wait=0`, I can see the error `Error sending message, child exited 127 (Exec error.)`.

I can see in `man 5 neomuttrc`

       sendmail_wait
              Type: number
              Default: 0

              Specifies the number of seconds to wait for the $sendmail process to finish before giving up and putting delivery in the background.

              NeoMutt interprets the value of this variable as follows:
              >0     number of seconds to wait for sendmail to finish before continuing
              0      wait forever for sendmail to finish
              <0     always put sendmail in the background without waiting

              Note that if you specify a value other than 0, the output of the child process will be put in a temporary file.  If there is some error, you will be informed as to where to find the output.

The last line suggests to me that I should be informed somehow of an error?

**Comment by roccoblues** [comment]
Yes, it's definitely related to `sendmail_wait=-1`.

I had a quick look at the code but I don't see how we can notify the user of an error if we don't wait for the process to end. Of course I'm no expert so it's possible I'm missing something. Maybe @flatcap knows more?
